java/org/apache/catalina/valves/AbstractAccessLogValve.java

```diff
@@ -1756,10 +1776,14 @@ protected AccessLogElement createAccessLogElement(String name, char pattern) {
                 return new DateAndTimeElement(name);
             case 'T':
                 // ms for milliseconds, us for microseconds, and s for seconds
-                if ("ms".equals(name)) {
-                    return new ElapsedTimeElement(false, true);
+                if ("ns".equals(name)) {
+                    return new ElapsedTimeElement(ElapsedTimeElement.Style.MILLISECONDS);
```

Comment:
I think you want `NANOSECONDS` here.

Reply:
If course I did. Thanks for spotting that.